### ISSUE TITLE
[SMN] Re-create SMN topic fails

### ISSUE THREAD

**Issue by outcatcher** [opentelekomcloud/terraform-provider-opentelekomcloud#610]
### Terraform Version
```
Terraform v0.13.1
+ provider registry.terraform.io/opentelekomcloud/opentelekomcloud v1.19.5
```

### Affected Resource(s)
`opentelekomcloud_smn_topic_v2`
`opentelekomcloud_smn_subscription_v2`

### Terraform Configuration Files
```hcl
resource "opentelekomcloud_smn_topic_v2" "alarms" {
  name         = "alarms2"
  display_name = "Alarms in Test2"
}

resource "opentelekomcloud_smn_subscription_v2" "almo-outage" {
  topic_urn = opentelekomcloud_smn_topic_v2.alarms.id
  protocol  = "email"
  endpoint  = "alarms@mycompany.com"
}

```

### Expected Behavior
Topic removed and created again

### Actual Behavior
```
Error: topic: Expected HTTP response code [202 204] when accessing [GET https://smn.eu-de.otc.t-systems.com/v2/5045c215010c440d91b2f7dce1f3753b/notifications/topics/urn:smn:eu-de:5045c215010c440d91b2f7dce1f3753b:alarms2], but got 200 instead
{"update_time":"2020-09-04T14:04:46Z","push_policy":0,"create_time":"2020-09-04T14:04:46Z","name":"alarms2","topic_urn":"urn:smn:eu-de:5045c215010c440d91b2f7dce1f3753b:alarms2","display_name":"Alarms in Test2","request_id":"BEC29FF227F34B2A817685D04E96B31E"}

```

### Steps to Reproduce
1. `terraform apply`
2. Change topic name
3. `terraform apply`

### References
First noticed in #607 by @almo-ltoeroek

**Comment by outcatcher** [comment]
Seems to be caused by shared `RequestOpts` in `smn/v2/topics/requests.go`

**Comment by outcatcher** [comment]
@almo-ltoeroek can't reproduce this issue after SDK migration

Can you please check it with `v1.21.0`?

**Comment by almo-ltoeroek** [comment]
@outcatcher Tested with original case with v1.21.0 a couple times, also with tweaks, seems fixed. I can't reproduce the issue any more.

**Comment by lego963** [comment]
@almo-ltoeroek it could be problems with API

**Comment by outcatcher** [comment]
@lego963 this was fixed in SDK before :) so #641 fixed this issue for the provider
